Set remotes in description

kiwiroy opened the pull request:
1. Update the `DESCRIPTION` file to refer to Bioconductor `Remotes` so that the install instructions work and minor typo.
2. Added GitHub Actions to test (1) and run `rcmdcheck`
3. Modify examples so `rmcdcheck` passes them.

kiwiroy committed:
fix minor typos, update readme, check for http errors, and add GH actions

kiwiroy committed:
fix minor typos, update readme, check for http errors, and add GH actions

FelixErnst commented on .github/workflows/R-CMD-check.yaml:
```diff
@@ -0,0 +1,31 @@
+      - name: Install dependencies
+        run: |
+          install.packages(c("remotes", "rcmdcheck"))
+          remotes::install_deps(dependencies = TRUE)
```
works fine with `remotes::install_deps(dependencies = TRUE, repos = BiocManager::repositories())`.

Using this, the changes in the DESCRIPTION file shouldn't be necessary

kiwiroy replied:
Done, `README.md` updated to reflect it.

FelixErnst commented on DESCRIPTION:
```diff
@@ -42,5 +42,9 @@ Collate:
   'RevigoR-plot.R'
   'RevigoR-utils.R'
   'RevigoRData-class.R'
+Remotes:
```
See comment above

kiwiroy replied:
Done.

FelixErnst commented on R/RevigoR-RDAVIDWebService.R:
```diff
@@ -17,10 +17,12 @@ NULL
 #' @export
 #'
 #' @examples
+#' \dontrun{
```
They don't work anymore? Why? It is not a good idea to include dontrun statements here

kiwiroy replied:
Yeah, they didn't, but now they do, looks like it was a combination of X11 error and website error. Added `http_error` test.

FelixErnst commented on R/RevigoR-plot.R:
```diff
@@ -29,10 +29,12 @@ NULL
 #' @importFrom ggthemes solarized_pal
 #'
 #' @examples
+#' \dontrun{
```
See comment above

kiwiroy replied:
As above

FelixErnst commented on .github/workflows/R-CMD-check.yaml:
```diff
@@ -0,0 +1,31 @@
+
+jobs:
+  R-CMD-check:
+    runs-on: macOS-latest
```
Please include a linux build or switch to linux entirely. Thanks

kiwiroy replied:
Done.

FelixErnst reviewed (changes requested):
Hi Roy @kiwiroy ,

thanks for the PR. Since I don't have the permission of the website hoster to put this in BioC, I am not actively developing here. 

However, If you fix the issues mentioned in the comments, I am happy to merge the PR:

Felix

kiwiroy commented:
@FelixErnst I resolved all the suggestions you've made. Revigo web service was not stable yesterday confounding my installation woes, hopefully I've made it a little more obvious when the service isn't responding.  Thanks for the codes.

FelixErnst committed:
cleanup and tweak
- assertive dependency removed

FelixErnst committed:
merged

Merge branch 'bioc-remotes' of https://github.com/kiwiroy/RevigoR into pr/4

# Conflicts:
#	.github/workflows/R-CMD-check.yaml
#	DESCRIPTION
